extension/public/popup.js

```diff
@@ -26,28 +23,36 @@ const globalSwitchHandler = () => {
     toggleList.style.pointerEvents = "auto";
     toggleList.style.opacity = "100%";
   }
-
-  chrome.storage.sync.set({ globalToggled: globalSwitch.checked });
 };
 
-const redditSwitchHandler = () => {
-  chrome.storage.sync.set({ redditToggled: redditSwitch.checked });
+// a map of all the switche elements, keyed by their storageKey in chrome storage
+const switches = {
+  redditToggled: redditSwitch,
+  instagramToggled: instagramSwitch,
+  twitterToggled: twitterSwitch,
+  globalToggled: globalSwitch,
 };
 
-// onchange instead of onclick as the state may change programatically
-globalSwitch.onchange = globalSwitchHandler;
-redditSwitch.onchange = redditSwitchHandler;
-
 // Set initial state of toggles
-chrome.storage.sync.get(["redditToggled"], (data) => {
-  redditSwitch.checked = data.redditToggled;
-  redditSwitchHandler();
-});
-
-chrome.storage.sync.get(["globalToggled"], (data) => {
-  globalSwitch.checked = data.globalToggled;
-  globalSwitchHandler();
-});
+for (const storageKey in switches) {
+  const switchEl = switches[storageKey];
```

Comment:
It wasn't immediately obvious to me that `switchEl` = switch elements. Perhaps rename the variable. If this is a convention you've seen in other projects, you can ignore this comment.

Reply:
good point, this is a bad variable name. I've changed it to be more descriptive